src/lib.rs

```diff
@@ -1,9 +1,59 @@
-//  FileName    : lib.rs
-//  Author      : ShuYu Wang <andelf@gmail.com>
-//  Created     : Tue Sep 13 23:37:09 2016 by ShuYu Wang
-//  Copyright   : Feather Workshop (c) 2016
-//  Description : A double array trie implementation.
-//  Time-stamp: <2016-09-13 23:38:14 andelf>
+//! Double Array Trie in Rust
+//!
+//! ## Installation
+//!
+//! Add it to your `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! darts = "0.1"
+//! ```
+//!
+//! then you are good to go. If you are using Rust 2015 you have to ``extern crate darts`` to your crate root as well.
+//!
+//! ## Example
+//!
+//! ```rust
+//! use darts::DoubleArrayTrie;
+//! use std::fs::File;
+//!
+//! fn main() {
+//!     let mut f = File::open("./priv/dict.big.bincode").unwrap();
+//!     let da = DoubleArrayTrie::load(&mut f).unwrap();
+//!     let string = "中华人民共和国";
+//!     let prefixes = da.common_prefix_search(string).as_ref().map(|matches| {
+//!         let res = matches
+//!             .iter()
+//!             .map(|&(end_idx, v)| {
+//!                 &string[..end_idx]
+//!             })
+//!             .collect();
+//!         res
+//!     }).unwrap_or(vec![]);
```

Comment:
let res = foo;
    res

is equivalent to

    foo

btw, is ``into_iter`` more suitable here?